### PR TITLE
Stylable badge

### PR DESCRIPTION
**Pull request by lbelinsk**
Changed the api a little bit.
The BO component will not have the "minWidth" as a mixin prop since imo it is too specific to the counterBadge.

The core shouldn't have the specific D20 color in it's palette, so i changed it to be grey.

### CODE REVIEW

File: packages/wix-ui-core/src/components/StylableBadge/Badge.driver.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+      exists: () => !!element,
+      /** returns elements type attribute */
+      getType: () => element.getAttribute('type'),
+      /** returns elements innerHtml */
+      getContent: () => element.innerHTML,
+      /** returns elements innerText */
+      getContentText: () => element.textContent,
+    };
+  };
+
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/StylableBadge/Badge.driver.ts]: need a new line

File: packages/wix-ui-core/src/components/StylableBadge/Badge.protractor.driver.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const badgeDriverFactory = component => ({
+    /** returns the component element */
+    element: () => component,
+    /** returns the component text */
+    getTextContent: () => component.getText(),
+  });
+
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/StylableBadge/Badge.protractor.driver.ts]: new line

File: packages/wix-ui-core/stories/StylableBadge/index.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import {Badge as StylableBadge} from '../../src/components/StylableBadge';
+import commonStyle from '../../src/components/StylableBadge/BadgeStyle.st.css';
+
+export class StylableBadgeStory extends React.Component<{}, {}> {
```
Comment (shlomitc) [inline packages/wix-ui-core/stories/StylableBadge/index.tsx]: I think that the AutoDocs tool works for it

Comment (lbelinsk) [reply packages/wix-ui-core/stories/StylableBadge/index.tsx]: I see that it is possible only for the core "reset" look, not for the default look if I am not wrong

Comment (shlomitc) [reply packages/wix-ui-core/stories/StylableBadge/index.tsx]: I think it’s good to have the core look.
In general, the suggested theme should only be an example and not the main thing.
You can add it as an additional example at the bottom of the automated story

Comment (shlomitc) [reply packages/wix-ui-core/stories/StylableBadge/index.tsx]: I think it’s good to have the core look.
In general, the suggested theme should only be an example and not the main thing.
You can add it as an additional example at the bottom of the automated story

**Pull request merged**